src/deadline/mcp/tools/utils.py

```diff
@@ -0,0 +1,201 @@
+    error_info = {
+        "error": str(e),
+        "type": type(e).__name__
+    }
+    
+    if hasattr(e, "response") and hasattr(e.response, "status_code"):
+        error_info["status_code"] = e.response.status_code
+    
+    logger.error(f"API tool error: {error_info}", exc_info=True)
+    return error_info
+
+
+def _is_telemetry_wrapped(sig: inspect.Signature) -> bool:
+    """Check if a function signature indicates telemetry wrapping (*args, **kwargs)."""
+    return (
+        len(sig.parameters) == 2 and
+        list(sig.parameters.keys()) == ['args', 'kwargs'] and
+        all(p.kind in [inspect.Parameter.VAR_POSITIONAL, inspect.Parameter.VAR_KEYWORD] 
+            for p in sig.parameters.values())
+    )
+
+
+def _create_aws_api_parameters(func_name: str) -> list[inspect.Parameter]:
+    """Create AWS API parameters based on function name patterns."""
+    # Common AWS Deadline API parameters
+    params = [
+        inspect.Parameter('dryRun', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        inspect.Parameter('nextToken', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        inspect.Parameter('principalId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        inspect.Parameter('maxResults', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        inspect.Parameter('studioId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+    ]
+    
+    # Add function-specific parameters
+    func_lower = func_name.lower()
+    if 'farm' in func_lower:
+        params.append(inspect.Parameter('farmId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str))
+    elif 'queue' in func_lower:
+        params.extend([
+            inspect.Parameter('farmId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+            inspect.Parameter('queueId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        ])
+    elif 'job' in func_lower:
+        params.extend([
+            inspect.Parameter('farmId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+            inspect.Parameter('queueId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+            inspect.Parameter('jobId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        ])
+    elif 'fleet' in func_lower:
+        params.extend([
+            inspect.Parameter('farmId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+            inspect.Parameter('fleetId', inspect.Parameter.KEYWORD_ONLY, default=None, annotation=str),
+        ])
+    
+    return params
+
+
+def _filter_kwargs(kwargs: dict) -> dict:
+    """Filter out None, empty string, and 'null' values from kwargs."""
+    return {k: v for k, v in kwargs.items() if v is not None and v != "" and v != "null"}
+
+
+def _create_telemetry_wrapper(original_func, func_name: str, tool_serializer, tool_error_handler):
```

Comment:
Why is this function so much smaller than the regular wrapper below? Seems like they should be similar? This function should just be ignoring the telemetry stuff right?